File: firstrun/gemma4-31b/gemma4_31b/file_003_naive_backend.js
const express = require('express');
const fs = require('fs').promises;
const path = require('path');
const app = express();
const PORT = 3000;

// Directory where log files are stored
const LOGS_DIR = path.join(__dirname, 'logs');

// Ensure logs directory exists and has some dummy data for demonstration
async function initLogs() {
    try {
        await fs.mkdir(LOGS_DIR, { recursive: true });
        const sampleFiles = ['system.log', 'error.log', 'access.log'];
        for (const file of sampleFiles) {
            const content = `Log entries for ${file}\n${new Date().toISOString()} - INFO - System initialized\n${new Date().toISOString()} - WARN - High memory usage detected`;
            await fs.writeFile(path.join(LOGS_DIR, file), content);
        }
    } catch (err) {
        console.error('Error initializing logs:', err);
    }
}

// Route to list all log files in the directory
app.get('/api/logs', async (req, res) => {
    try {
        const files = await fs.readdir(LOGS_DIR);
        const logFiles = files.filter(file => file.endsWith('.log'));
        res.json(logFiles);
    } catch (err) {
        res.status(500).json({ error: 'Could not read logs directory' });
    }
});

// Route to get the content of a specific log file
app.get('/api/logs/:filename', async (req, res) => {
    try {
        const filename = req.params.filename;
        
        // Security: Prevent path traversal attacks
        if (filename.includes('..') || filename.includes('/') || filename.includes('\\')) {
            return res.status(403).send('Invalid filename requested');
        }

        const filePath = path.join(LOGS_DIR, filename);
        const data = await fs.readFile(filePath, 'utf8');
        res.setHeader('Content-Type', 'text/plain');
        res.send(data);
    } catch (err) {
        if (err.code === 'ENOENT') {
            res.status(404).send('Log file not found');
        } else {
            res.status(500).send('Error reading log file');
        }
    }
});

// Serve the HTML page
app.get('/', (req, res) => {
    res.sendFile(path.join(__dirname, 'index.html'));
});

initLogs().then(() => {
    app.listen(PORT, () => {
        console.log(`Log viewer running at http://localhost:${PORT}`);
    });
});

File: firstrun/qwen3.6:27b/qwen3_6_27b/file_003_naive_backend.py
import os
from flask import Flask, request, jsonify

app = Flask(__name__)

# Define the directory where log files are stored
LOG_DIR = 'logs'

@app.route('/api/logs/<filename>', methods=['GET'])
def get_log_file(filename):
    # Prevent directory traversal attacks by sanitizing the filename
    if '..' in filename or '/' in filename or '\\' in filename:
        return jsonify({'error': 'Invalid filename'}), 400
    
    # Construct the full path to the log file
    file_path = os.path.join(LOG_DIR, filename)
    
    # Check if the file exists and is a file (not a directory)
    if not os.path.isfile(file_path):
        return jsonify({'error': 'File not found'}), 404
    
    try:
        with open(file_path, 'r') as f:
            content = f.read()
        return jsonify({'content': content}), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500

if __name__ == '__main__':
    # Ensure the log directory exists
    os.makedirs(LOG_DIR, exist_ok=True)
    app.run(debug=True)